### PR TITLE
Dataflow API

### PR DESCRIPTION
**Pull request by dhruvigoyal25**
This PR makes the following changes to AdvantEDGE:
1. Added a new endpoint in swagger for Dataflow API
2. Added backend code for Dataflow API

### CODE REVIEW

File: go-apps/meep-metrics-engine/api/v2/swagger.yaml [hunk 1]
```diff
@@ -426,6 +451,89 @@ definitions:
...
+      tags:
+        type: "array"
+        description: "Tag names to match in query. Supported values:<br> "
+        items:
+          $ref: "#/definitions/Tag"
+      scope:
+        $ref: "#/definitions/Scope"
+      fields:
+        type: "array"
+        description: "Requested information. Supported values:<br> "
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/api/v2/swagger.yaml:466]: Add list of supported fields + update example accordingly.

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 2]
```diff
@@ -719,6 +719,105 @@ func updateSeqMetrics(log string, metric SeqMetric, params SeqQueryParams, respo
...
+	if metricStore == nil {
+		err := errors.New("No active scenario to get metrics from")
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+
+	// Make sure only type of format is specified
+	if len(params.Fields) > 1 {
+		err := errors.New("Specify only one type of format: meraid or sdorg")
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/v2/metrics-engine.go:752]: sdorg is not a supported field.

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 3]
```diff
@@ -719,6 +719,105 @@ func updateSeqMetrics(log string, metric SeqMetric, params SeqQueryParams, respo
...
+		log.Error(err.Error())
+		http.Error(w, err.Error(), http.StatusNoContent)
+		return
+	}
+
+	// Prepare & send response
+	var response DataflowMetrics
+	response.DataflowMetricList = &DataflowMetricList{
+		Name:    "sequence metrics",
+		Columns: append(params.Fields, "time"),
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/v2/metrics-engine.go:792]: Fields list should only include time & mermaid, is this the case here?

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 4]
```diff
@@ -719,6 +719,105 @@ func updateSeqMetrics(log string, metric SeqMetric, params SeqQueryParams, respo
...
+	}
+	interactions := make(map[string]int)
+	for i := len(valuesArray) - 1; i >= 0; i-- {
+		values := valuesArray[i]
+		var metric DataflowMetric
+		metric.Time = values["time"].(string)
```
Comment (dilallkx) [inline go-apps/meep-metrics-engine/server/v2/metrics-engine.go:798]: Use the "time" constant that should already be defined in the metrics package.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go (hunk 3 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go:792]: Corrected this to only show requested fields and time

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go (hunk 2 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go:752]: Corrected the error message.

File: go-apps/meep-metrics-engine/api/v2/swagger.yaml (hunk 1 above)
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/api/v2/swagger.yaml:466]: Addressed.

**Review by dilallkx — CHANGES_REQUESTED** [review]

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 5]
```diff
@@ -719,6 +721,106 @@ func updateSeqMetrics(log string, metric SeqMetric, params SeqQueryParams, respo
...
+	limit := 0
+	if params.Scope != nil {
+		duration = params.Scope.Duration
+		limit = int(params.Scope.Limit)
+	}
+	// Get http metrics
+	var response DataflowMetrics
+	response.DataflowMetricList = &DataflowMetricList{
+		Name:    "sequence metrics",
+		Columns: append(params.Fields, "time"),
```
[the remark this answers is not in the document]
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go]: Updated

File: go-apps/meep-metrics-engine/server/v2/metrics-engine.go [hunk 6]
```diff
@@ -561,6 +561,16 @@ func mePostSeqQuery(w http.ResponseWriter, r *http.Request) {
 		duration = params.Scope.Duration
 		limit = int(params.Scope.Limit)
 	}
+
+	var response SeqMetrics
+	mobilityEventProcessed := false
+	if params.ResponseType == listOnly || params.ResponseType == "" {
+		response.SeqMetricList = &SeqMetricList{
+			Name:    "sequence metrics",
+			Columns: append(params.Fields, "time"),
```
[the remark this answers is not in the document]
Comment (dhruvigoyal25) [reply go-apps/meep-metrics-engine/server/v2/metrics-engine.go]: Updated

**Review by dilallkx — APPROVED** [review]

File: .meepctl-repocfg.yaml [hunk 7]
```diff
@@ -38,7 +38,7 @@ repo:
     # platform ingress configuration
     ingress:
       # host name
-      host: my-platform-fqdn
+      host: 10.3.16.69
```
[the remark this answers is not in the document]
Comment (dilallkx) [reply .meepctl-repocfg.yaml]: Fixed.

**Pull request merged**